mrmustard/utils/training.py

```diff
@@ -12,17 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from logging import Logger
-import numpy as np
-from scipy.linalg import expm
-from mrmustard.types import *
+"""This module contains the implementation of optimization classes and functions
+used within Mr Mustard.
+"""
+
+from mrmustard.types import List, Callable, Sequence, Trainable, Tensor
 from mrmustard.utils import graphics
 from mrmustard.logger import create_logger
 from mrmustard.math import Math
 
 math = Math()
 
-
+# pylint: disable=disallowed-name
```

Comment:
👀

Reply:
A python programmer walks into a `bar` and asks for something imported 😅